### DISCUSSION TITLE
Phalcon Logger Interpolation does not work

### DISCUSSION THREAD

**Discussion by gnikolopoulos** [phalcon/cphalcon#16425] [Q&A]
Hi all,

I am using the code from [here](https://docs.phalcon.io/5.0/en/logger#interpolation) in order to inject additional, dynamic data into my logging process while calling an external API, but I only get the placeholders themselves instead of the context values.

I am using Phalcon 5.2.1 and even tried the code on the example as is, to no avail.

**Comment by s-ohnishi** [comment]
I tested it and got the same result.
I checked the source of .zep and finally found this.
```PHP
Phalcon\Support\Helper\Str\AbstractStr::toInterpolate(
    string input,
    array context = [],
    string left = "%",
    string right = "%"
)
```
And neither `left` and `right` are specified when calling, so the description in the manual is wrong.
Following this will give you the expected result:
```PHP
$message = '%framework% executed the "Hello World" test in %secs% second(s)';
```

**Reply by s-ohnishi** [reply]
An inconsistency was found in the [HelperFactory.zep](https://github.com/phalcon/cphalcon/blob/master/phalcon/Support/HelperFactory.zep) description.
Show only the problem part.
````
  * @method bool between(int $value, int $start, int $end)
````
and
```PHP
  "filter" : "Phalcon\\Support\\Helper\\Arr\\Filter",
  "isBetween" : "Phalcon\\Support\\Helper\\Number\\IsBetween",
  "dynamic" : "Phalcon\\Support\\Helper\\Str\\Dynamic",
  "interpolate" : "Phalcon\\Support\\Helper\\Str\\Interpolate",
````
"isBetween" is "between" in PHPDoc, and nothing else exists.

**Reply by niden** [reply]
Thank you @s-ohnishi I fixed that in the latest PR

**Reply by s-ohnishi** [reply]
The test I was taught was "not found".
Is it something that hasn't been published or pushed?

I think it's a problem that it can't be referenced properly in the IDE, so I hope the PR will be merged soon.

**Reply by niden** [reply]
The PR was merged and available with the new release. Can you check if you don't mind?

**Reply by s-ohnishi** [reply]
I only have a rough understanding of Zephir, so even if I look at the commit, I don't understand it very well, so I decline.
However, it seems that the comment part affects ide-stubs, so I would like you to correct the comment in HelperFactory.zep.

**Comment by Jeckerson** [comment]
Issue created https://github.com/phalcon/cphalcon/issues/16427
